app/controllers/activity_forms_controller.rb

```diff
@@ -74,6 +77,12 @@ def update
       return
     end
 
+    if step == :implementing_organisation && @activity.errors.present?
+      @implementing_organisations = Organisation.active.sorted_by_name
+      render_step :implementing_organisation
+      return
+    end
```

Comment:
Is this essentially "repeat the step if there are errors"? Although I think all the steps do that by default, so maybe not. Is there a way of clarifying what this is doing/why it doesn't live inside the `case` statement, perhaps by abstracting into a method?